velox/common/memory/SharedArbitrator.cpp

```diff
@@ -281,10 +282,22 @@ uint64_t SharedArbitrator::decrementFreeCapacityLocked(
   return allocatedBytes;
 }
 
+uint64_t SharedArbitrator::adjustedShrinkCapacity(
+    const MemoryPool& pool,
+    uint64_t targetBytes) const {
+  const auto reservedBytes = pool.reservedBytes();
+  const auto headroomBytes =
+      std::min(static_cast<uint64_t>(reservedBytes / 4), 128ULL << 20);
+  const auto adjustedTargetBytes =
```

Comment:
```
const int64_t adjustedTargetBytes = std::max(0, pool.capacity() - (reservedBytes + headroomBytes));
```
Can you have a test case to cover the case reservedBytes + headroomBytes > pool.capacity() to make sure the integer overflow is handled properly? Thanks!